tsdb/executor.go

```diff
@@ -157,10 +157,18 @@ func (e *Executor) executeRaw(out chan *influxql.Row) {
 		}
 	}
 
-	// Get the union of SELECT fields across all mappers.
-	selectFields := newStringSet()
-	for _, m := range e.mappers {
-		selectFields.add(m.Fields()...)
+	// Get the distinct fields across all mappers.
+	var selectFields, aliasFields []string
+	if e.stmt.HasWildcard() {
+		sf := newStringSet()
```

Comment:
For `select *` a `stringSet` works perfect as it gets you the distinct fields sorted alphabetically.  It's when it wasn't a `select *` that the problem was encountered, and what this PR fixes.

`select mean(value), max(value) from foo` scenario works and is tested with this test:

https://github.com/influxdb/influxdb/pull/3564/files#diff-09d8dc75f001e1e2a64439ee67dd14cbR1062